readme.txt
----- How to install -----

	IMPORTANT NOTE: This is not updated for Poschengband R. 

--- LINUX
  Download and unpack the source archive, or clone the git repository:
    $ git clone https://github.com/poschengband/poschengband.git

  Make sure you have the appropriate development packages installed.
  For example, you might run (Ubuntu or Mint):
    $ sudo apt-get install autoconf gcc libc6-dev libncursesw5-dev libx11-dev

  From the root of the source archive:
    $ sh autogen.sh
    $ ./configure
    $ make clean
    $ make

  To install, you may need to elevate your credentials:
    $ su
    $ make install
    $ exit

  Then run poschengband as desired:
    $ poschengband -- -n<number of windows>  ## for normal ASCII graphics
  or
    $ poschengband -g -- -n<# of windows>    ## for 8x8 tile graphics 

  You can change game windows' font, location, and size, by environment 
  variables.

  Ex.
    $ set env ANGBAND_X11_FONT '-*-*-medium-r-normal--24-*-*-*-*-*-iso8859-1'
    $ poschengband -- -n

  Then font size will be changed.

  You can set ANGBAND_X11_FONT_n for specific window which have window number n.

  Location of windows are ANGBAND_X11_AT_X_n, and ANGBAND_X11_AT_Y_n.
  Size of windows are ANGBAND_X11_COLS_n, and ANGBAND_X11_ROWS_n.

  (Thanks to Nick McConnell for implementing and improving building under Linux/MAC!)

  (ALT: For development, doing an install is undesirable. Try the following instead:
    $ sudo apt-get install clang-3.5 llvm-3.5
    $ sh autogen.sh
    $ ./configure CFLAGS=-fsanitize=address LDFLAGS=-fsanitize=address --with-no-install CC=clang-3.5
    $ make clean
    $ make -j4
    $ cp src/poschengband .
    $ ASAN_SYMBOLIZER_PATH=/usr/bin/llvm-symbolizer-3.5 ./poschengband -g -u<Savefile> -- -n1
    
    Note, you need to remember to copy the executable up after every build (TODO: Fix Makefile to do this!)
    Also, I like to develop with the address sanitizer enabled at all times. This works best with clang as
    your compiler, rather than gcc. Of course, 3.5 should be replaced with latest version of clang.).

    ***Fonts on Linux: My experience with linux has been that the fonts are just plain awful. Here, I document
    how I managed to install better fonts and use them in PosChengband. I spent nearly a day on this, so I'm
    hoping this might prove useful to somebody else. But mostly, it is here to remind me how I did so I don't
    stumble so much next time.

    [1] Install some better fonts. For example
    $ sudo apt-get install fonts-liberation

    [2] See what fonts are on your system that PosChengband can use:
    $ xlsfonts
    Notice that the new fonts aren't there!

    [3] Find where the new fonts were installed. For example
    $ fc-list | grep liberation
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Italic.ttf: Liberation Sans Narrow:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf: Liberation Sans:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationMono-BoldItalic.ttf: Liberation Mono:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-Italic.ttf: Liberation Serif:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationMono-Bold.ttf: Liberation Mono:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Regular.ttf: Liberation Sans Narrow:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSerif-Bold.ttf: Liberation Serif:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationMono-Regular.ttf: Liberation Mono:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSans-Italic.ttf: Liberation Sans:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-BoldItalic.ttf: Liberation Serif:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-BoldItalic.ttf: Liberation Sans Narrow:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationMono-Italic.ttf: Liberation Mono:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSans-BoldItalic.ttf: Liberation Sans:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-Regular.ttf: Liberation Serif:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Bold.ttf: Liberation Sans Narrow:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationSans-Bold.ttf: Liberation Sans:style=Bold

    [4] Bang head on wall in frustration!!

    [5] Tell X Server about the fonts:
    $ cd /usr/share/fonts/truetype/liberation/
    $ su
    $ mkfontscale
    $ mkfontdir
    $ exit
    $ cat fonts.dir
    ...
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-adobe-standard
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-ascii-0
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-ibm-cp437
    ... [and many more]

    $ xset +fp /usr/share/fonts/truetype/liberation
    $ xset fp rehash
    $ xlsfonts | grep liberation
    ...
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-1
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-10
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-13
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-15
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-16
    ... [and many more]

    [6] Now, try to find a font you like. For example:
    $ cd [path to poschengband]
    ~/Src/poschengband> ANGBAND_X11_FONT='-misc-liberation mono-medium-r-normal--20-0-0-0-m-0-iso8859-1' ./poschengband -mx11
    You can play with the point size since these are vector fonts. I chose 20pt since my eyes suck!

    [7] Smile. Grab beer to recover from step [4].

--- Curses

  Curses is for Linux, of course, so everything said above also applies here.
  To run with a single 'big' terminal, simply run, for example:
    $ ./poschengband -mgcu -uCrusher

  To add additional terminal windows, you need to specify sub-options. You can
  configure, from the command line, a strip of terminals on the right hand side
  of the screen, or on the bottom of the screen, or both.

  For example:
    $ ./poschengband -mgcu -uCrusher -- -right 57x26,*

  This specifies that the right hand strip will be be 57 columns wide, and will
  contain 2 additional terminals. The first one, on top, will be 57x26 (i.e., 26
  rows high) while the second terminal will be 57x(LINES-26) (i.e., * means fill
  to whatever is leftover). These terminals will be numbered 1 and 2, respectively.

  Another example:
    $ ./poschengband -mgcu -uCrusher -- -bottom -bottom *x10

  This adds a bottom strip 10 rows high, and this strip will contain a single
  additional terminal window (numbered as Term-1) that will be as wide as the
  screen allows (i.e, Term-1 will be COLSx10).

  Finally, you can combine the -right and -bottom commands, in either order.
  For example:
    $ ./poschengband -mgcu -uCrusher -- -right 57x26,* -bottom *x10

  Here, Term-1 and Term-2 are on the right strip, sized as 57x26 and 57x(LINES-26),
  while Term-3 is on the bottom strip, sized (COLS-57)x10. Term-0, the Main Terminal,
  is, of course, located in the top left and sized as (COLS-57)x(LINES-10). The
  map terminal will always use as much space as possible.

  Finally, the order of the -right and -bottom command is significant, as it affects
  both the child terminal numbering and also resolves the conflict over the overlap
  region in the bottom-right corner of the screen.

  For example, the meaning of the following should now be clear:
    $ ./poschengband -mgcu -uCrusher -- -bottom *x10 -right 57x26,*

  You cannot specify more than 7 child terminals.

--- Windows

  Download the binary archive for Windows.  Unzip it to any location that you 
  will have full permissions and launch poschengband to play.

  To compile, download the poschengband source code and unzip. I currently use
  Visual C++ 2010 Express which is freely available. I tried and don't recommend
  the 2012 Express edition as the MS seems to have "broken" Intellisense once
  again. Open poschengband.sln and compile.


-----  Basic for Playing  ------

  I've done quite a bit of work on the in-game documentation, though it never
  seems to be a completable task. If you are new to the game, I recommend you
  read the Newbie Guide (Press ? to activate the help system, then choose a for
  General Information followed by e). This will give a quick tutorial as well
  as some additional information about the game. Helpfiles are also available
  as html files for viewing outside the game in your favorite browser. In this
  case, open lib/help/html/tang.html.

---------  Commands  ----------- 

  Again, please refer to the in-game help (It hurts me to keep the same information
  up to date in multiple places). Press ? for help and then k for Commands. Choose
  the topic that interests you (Command Descriptions for an overview, or one of the
  Keyset options for reference on which keys to press).

  Or open lib/help/html/commdesc.html or lib/help/html/command.html in your browser.



changelog.txt
Changelog (from "vanilla" poschengband, as of Feb 10, 2017, with commit hashes
           in parentheses)

Significant gameplay changes:
-----------------------------
- New monsters by clouded. (0c2980e)
- Add an option to only replace half of all fixedarts with randarts. (aec1a15)
- Remove non-rune spells from rune-knights, remove MP costs from rune spells.
  (a7754bd, 926bae6, 1526f52)

Minor gameplay changes:
-----------------------
no_wilderness changes:
- Give dungeon guardians their special drop in no_wilderness. (bd8e73c)
- Add warg quest to t_lite. (6041335)
- Remove WILD_ONLY from two uniques (so that they generate in no_wilderness).
  (bfd337a)

randart changes:
- Keep activations of replaced randarts. (27716f8, 425a0eb, 151d683, bc0a7c9)
- Tweak replacementart object_level. (81ad7bc)
- Tweak randart rings. (e1321a2)

specific races/classes:
- Make !newlife reroll subrace for those monster races with random subrace
  (a2907a6, 2a14d0d)
- Remove erratic movement for players. (c3d6de5)
- Make Duelists automatically duel when attacking a monster. (32c911e)

other:
- Only trigger eldritch horror/fear effects for monsters in LOS (a150551)
- Change Warning to apply only to traps. (334a291)

Interface changes:
------------------
birth/town:
- Use the previous game's race/class/etc as default (50b0eff, 6ddbb80)
- Don't randomize starting gold. (7d1b8fa)
- Don't randomize starting object quantities. (5ce8ea8)
- Let males be sexy and females be lucky (7d1b8fa)
- Start with a lantern instead of torches (7d1b8fa)
- Reimplement no_selling option, but without giving larger gold piles. (74ac340)
- Allow all species to eat at the inn, don't fail to eat if already full.
  (b4aa846, 9421be4)
- Give Beginner spell proficiency immediately upon casting the spell once
  (since you can get to Beginner with 15 casts in town anyway). (12be6d5)

easy_lore (not all changes implemented properly for games with easy_lore
           turned off, since I always use it):
- Show more information (spell damage, EAT_WALL, etc) in monster recall.
  (dc853ff, a6c0a2d, 93ab4c0)
- Revert "Monster Lore: Always display the observed spell frequency" - at
  least in easy_lore this commit isn't good. (3be7781)
- Give "The X resists." messages for monsters that are only seen via telepathy
  (96d441b)
- Give players auto_id at XL 25 when using easy_lore. (5a2b4b8)

other:
- Cap idle time at 30s. (681321c)
- Automatic character dump on death (e79b7c5, 92c6322)
- Don't print a message when autosaving (2a0bc63)
- Don't give a message when failing to autodestroy artefacts (5ac422f)
- Always allow browsing spell information (6cbc176)
- Don't use special player char for monster species (this should probably
  actually be an option, since I know some people do like it). (e27e9fe)
- Don't reduce weaponsmith essence gain due to removable curses. (f687d74)
- Change landmines/lurkers etc to the 'x' glyph (4939c3c)
- Remove maprot from the Labyrinth (dc9fc7b)
- Let Ninjas regain stealth by spending a turn stationary. (498aba1)

Bug fixes:
----------
- Fix branding spells working on randarts. (c233c82)
- Fix enchanting prices not being additive. (8636a03)
- Fix targetting weirdnesses. (c2a63f4, 37791f0)
- Sync some quest rewards in t_lite.txt. (2f22904)
- Fix aether vortex damage display. (7addefc)
- Fix some int -> bool casts. (cd5b2cc)
- Fix shoggoth speech. (45b7da5)


src/mut.h
/* New Mutation System, carefully crafted to be backwards compatible with the old ... 
   You are free to add new mutations (though extending MUT_FLAG_SIZE requires save file
   work), and place activatable mutations where ever you please.
*/
#define MUT_SPIT_ACID      0   /* Start of old mutations */
#define MUT_BR_FIRE        1
#define MUT_HYPN_GAZE      2
#define MUT_TELEKINESIS    3
#define MUT_TELEPORT       4
#define MUT_MIND_BLAST     5
#define MUT_RADIATION      6
#define MUT_VAMPIRISM      7
#define MUT_SMELL_METAL    8
#define MUT_SMELL_MONSTERS 9
#define MUT_BLINK          10
#define MUT_EAT_ROCK       11
#define MUT_SWAP_POS       12
#define MUT_SHRIEK         13
#define MUT_ILLUMINE       14
#define MUT_DET_CURSE      15
#define MUT_BERSERK        16
#define MUT_POLYMORPH      17
#define MUT_MIDAS_TOUCH    18
#define MUT_GROW_MOLD      19
#define MUT_RESIST         20
#define MUT_EARTHQUAKE     21
#define MUT_EAT_MAGIC      22
#define MUT_WEIGH_MAGIC    23
#define MUT_STERILITY      24
#define MUT_PANIC_HIT      25
#define MUT_DAZZLE         26
#define MUT_LASER_EYE      27
#define MUT_RECALL         28
#define MUT_BANISH         29
#define MUT_COLD_TOUCH     30
#define MUT_LAUNCHER       31

#define MUT_BERS_RAGE      32
#define MUT_COWARDICE      33
#define MUT_TELEPORT_RND   34
#define MUT_ALCOHOL        35
#define MUT_HALLUCINATION  36
#define MUT_FLATULENT      37
#define MUT_SCORPION_TAIL  38
#define MUT_HORNS          39
#define MUT_BEAK           40
#define MUT_ATTRACT_DEMON  41
#define MUT_PROD_MANA      42
#define MUT_SPEED_FLUX     43
#define MUT_BANISH_ALL_RND 44
#define MUT_EAT_LIGHT      45
#define MUT_TRUNK          46
#define MUT_ATTRACT_ANIMAL 47
#define MUT_TENTACLES      48
#define MUT_RAW_CHAOS      49
#define MUT_NORMALITY      50
#define MUT_WRAITH         51
#define MUT_POLY_WOUND     52
#define MUT_WASTING        53
#define MUT_ATTRACT_DRAGON 54
#define MUT_RANDOM_TELEPATHY 55
#define MUT_NAUSEA         56
#define MUT_CHAOS_GIFT     57
#define MUT_SHADOW_WALK    58
#define MUT_WARNING        59
#define MUT_INVULN         60
#define MUT_SP_TO_HP       61
#define MUT_HP_TO_SP       62
#define MUT_FUMBLING       63

#define MUT_HYPER_STR      64
#define MUT_PUNY           65
#define MUT_HYPER_INT      66
#define MUT_MORONIC        67
#define MUT_RESILIENT      68
#define MUT_XTRA_FAT       69
#define MUT_ALBINO         70
#define MUT_FLESH_ROT      71
#define MUT_SILLY_VOICE    72
#define MUT_BLANK_FACE     73
#define MUT_ILL_NORM       74
#define MUT_XTRA_EYES      75
#define MUT_MAGIC_RES      76
#define MUT_XTRA_NOISE     77
#define MUT_INFRAVISION    78
#define MUT_XTRA_LEGS      79
#define MUT_SHORT_LEG      80
#define MUT_ELEC_AURA      81
#define MUT_FIRE_AURA      82
#define MUT_WARTS          83
#define MUT_SCALES         84
#define MUT_STEEL_SKIN     85
#define MUT_WINGS          86
#define MUT_FEARLESS       87
#define MUT_REGEN          88
#define MUT_ESP            89
#define MUT_LIMBER         90
#define MUT_ARTHRITIS      91
#define MUT_BAD_LUCK       92
#define MUT_VULN_ELEM      93
#define MUT_MOTION         94
#define MUT_GOOD_LUCK      95   /* End of the old mutations */

#define MUT_FAST_LEARNER   96
#define MUT_WEAPON_SKILLS  97
#define MUT_SUBTLE_CASTING 98
#define MUT_PEERLESS_SNIPER 99
#define MUT_UNYIELDING     100
#define MUT_AMBIDEXTROUS   101
#define MUT_UNTOUCHABLE    102
#define MUT_LOREMASTER     103
#define MUT_ARCANE_MASTERY 104
#define MUT_EVASION        105
#define MUT_POTION_CHUGGER 106
#define MUT_ONE_WITH_MAGIC 107
#define MUT_PEERLESS_TRACKER 108
#define MUT_INFERNAL_DEAL 109
#define MUT_MERCHANTS_FRIEND 110
#define MUT_SACRED_VITALITY 111
#define MUT_CULT_OF_PERSONALITY 112
#define MUT_FLEET_OF_FOOT 113
#define MUT_ASTRAL_GUIDE 114
#define MUT_DEMONIC_GRASP 115
#define MUT_WEIRD_MIND 116
#define MUT_FANTASTIC_FRENZY 117
#define MUT_BLACK_MARKETEER 118
#define MUT_SPEED_READER 119
#define MUT_FELL_SORCERY 120
#define MUT_TREAD_SOFTLY 121

#define MUT_DRACONIAN_SHIELD 122
#define MUT_DRACONIAN_MAGIC_RES 123
#define MUT_DRACONIAN_STRIKE 124
#define MUT_DRACONIAN_BREATH 125
#define MUT_DRACONIAN_REGEN 126
#define MUT_DRACONIAN_KIN 127
#define MUT_DRACONIAN_LORE 128
#define MUT_DRACONIAN_RESISTANCE 129
#define MUT_DRACONIAN_METAMORPHOSIS 130

#define MAX_MUTATIONS      131   /* see also MUT_FLAG_SIZE in defines.h */

typedef enum {
    MUT_RATING_AWFUL = -2,
    MUT_RATING_BAD   = -1,
    MUT_RATING_AVERAGE = 0,
    MUT_RATING_GOOD = 1,
    MUT_RATING_GREAT = 2
} mutation_rating;

typedef enum {
    MUT_TYPE_ACTIVATION = 0x01,    /* Mutation activates as a racial power (e.g. Harden to Elements) */
    MUT_TYPE_EFFECT = 0x02,     /* Mutation has a passive effect (e.g. Flatulence)*/
    MUT_TYPE_BONUS = 0x04,      /* Mutation confers a passive bonus (e.g. +4 Str) */
} mutation_type;

typedef bool (*mut_pred)(int mut_idx);
extern bool mut_good_pred(int mut_idx);
extern bool mut_bad_pred(int mut_idx);
extern bool mut_human_pred(int mut_idx);
extern bool mut_demigod_pred(int mut_idx);
extern bool mut_draconian_pred(int mut_idx);
extern bool mut_unlocked_pred(int mut_idx);

extern void mut_calc_bonuses(void);
extern void mut_get_flags(u32b flgs[TR_FLAG_ARRAY_SIZE]);
extern void mut_calc_stats(s16b stats[MAX_STATS]);
extern int  mut_gain_random_aux(mut_pred pred);
extern int  mut_count(mut_pred pred);
extern void mut_do_cmd_knowledge(void);
extern void mut_display(doc_ptr doc);
extern bool mut_gain(int mut_idx);
extern int mut_gain_choice(mut_pred pred);
extern bool mut_gain_random(mut_pred pred);
extern int  mut_get_powers(spell_info* spells, int max);
extern void mut_lock(int mut_idx);
extern bool mut_locked(int mut_idx);
extern bool mut_lose(int mut_idx);
extern void mut_lose_all(void);
extern bool mut_lose_random(mut_pred pred);
extern void mut_name(int i, char* buf);
extern void mut_help_desc(int i, char* buf);
extern bool mut_present(int mut_idx);
extern void mut_process(void);
extern int  mut_rating(int mut_idx);
extern int  mut_regenerate_mod(void);
extern int  mut_type(int mut_idx);
extern void mut_unlock(int mut_idx);

/* Non-spell mutations ... */
extern void albino_mut(int cmd, variant *res);
extern void alcohol_mut(int cmd, variant *res);
extern void ambidexterity_mut(int cmd, variant *res);
extern void arcane_mastery_mut(int cmd, variant *res);
extern void arthritis_mut(int cmd, variant *res);
extern void astral_guide_mut(int cmd, variant *res);
extern void attract_animal_mut(int cmd, variant *res);
extern void attract_demon_mut(int cmd, variant *res);
extern void attract_dragon_mut(int cmd, variant *res);
extern void bad_luck_mut(int cmd, variant *res);
extern void black_marketeer_mut(int cmd, variant *res);
extern void beak_mut(int cmd, variant *res);
extern void berserk_rage_mut(int cmd, variant *res);
extern void blank_face_mut(int cmd, variant *res);
extern void chaos_deity_mut(int cmd, variant *res);
extern void cowardice_mut(int cmd, variant *res);
extern void cult_of_personality_mut(int cmd, variant *res);
extern void eat_light_mut(int cmd, variant *res);
extern void demonic_grasp_mut(int cmd, variant *res);

extern void draconian_breath_mut(int cmd, variant *res);
extern void draconian_kin_mut(int cmd, variant *res);
extern void draconian_lore_mut(int cmd, variant *res);
extern void draconian_magic_resistance_mut(int cmd, variant *res);
extern void draconian_metamorphosis_mut(int cmd, variant *res);
extern void draconian_regen_mut(int cmd, variant *res);
extern void draconian_resistance_mut(int cmd, variant *res);
extern void draconian_shield_mut(int cmd, variant *res);
extern void draconian_strike_mut(int cmd, variant *res);

extern void einstein_mut(int cmd, variant *res);
extern void elec_aura_mut(int cmd, variant *res);
extern void evasion_mut(int cmd, variant *res);
extern void extra_eyes_mut(int cmd, variant *res);
extern void extra_legs_mut(int cmd, variant *res);
extern void extra_noise_mut(int cmd, variant *res);
extern void fantastic_frenzy_mut(int cmd, variant *res);
extern void fast_learner_mut(int cmd, variant *res);
extern void fat_mut(int cmd, variant *res);
extern void fearless_mut(int cmd, variant *res);
extern void fell_sorcery_mut(int cmd, variant *res);
extern void fire_aura_mut(int cmd, variant *res);
extern void flatulence_mut(int cmd, variant *res);
extern void fleet_of_foot_mut(int cmd, variant *res);
extern void fumbling_mut(int cmd, variant *res);
extern void good_luck_mut(int cmd, variant *res);
extern void hallucination_mut(int cmd, variant *res);
extern void he_man_mut(int cmd, variant *res);
extern void horns_mut(int cmd, variant *res);
extern void illusion_normal_mut(int cmd, variant *res);
extern void infernal_deal_mut(int cmd, variant *res);
extern void infravision_mut(int cmd, variant *res);
extern void invulnerability_mut(int cmd, variant *res);
extern void limber_mut(int cmd, variant *res);
extern void loremaster_mut(int cmd, variant *res);
extern void magic_resistance_mut(int cmd, variant *res);
extern void merchants_friend_mut(int cmd, variant *res);
extern void moron_mut(int cmd, variant *res);
extern void motion_mut(int cmd, variant *res);
extern void nausea_mut(int cmd, variant *res);
extern void normality_mut(int cmd, variant *res);
extern void one_with_magic_mut(int cmd, variant *res);
extern void peerless_sniper_mut(int cmd, variant *res);
extern void peerless_tracker_mut(int cmd, variant *res);
extern void polymorph_wounds_mut(int cmd, variant *res);
extern void potion_chugger_mut(int cmd, variant *res);
extern void produce_mana_mut(int cmd, variant *res);
extern void puny_mut(int cmd, variant *res);
extern void random_banish_mut(int cmd, variant *res);
extern void random_teleport_mut(int cmd, variant *res);
extern void random_telepathy_mut(int cmd, variant *res);
extern void raw_chaos_mut(int cmd, variant *res);
extern void regeneration_mut(int cmd, variant *res);
extern void resilient_mut(int cmd, variant *res);
extern void rotting_flesh_mut(int cmd, variant *res);
extern void sacred_vitality_mut(int cmd, variant *res);
extern void scales_mut(int cmd, variant *res);
extern void scorpion_tail_mut(int cmd, variant *res);
extern void shadow_walk_mut(int cmd, variant *res);
extern void short_legs_mut(int cmd, variant *res);
extern void silly_voice_mut(int cmd, variant *res);
extern void speed_flux_mut(int cmd, variant *res);
extern void speed_reader_mut(int cmd, variant *res);
extern void steel_skin_mut(int cmd, variant *res);
extern void subtle_casting_mut(int cmd, variant *res);
extern void telepathy_mut(int cmd, variant *res);
extern void tentacles_mut(int cmd, variant *res);
extern void tread_softly_mut(int cmd, variant *res);
extern void trunk_mut(int cmd, variant *res);
extern void untouchable_mut(int cmd, variant *res);
extern void unyielding_mut(int cmd, variant *res);
extern void vulnerability_mut(int cmd, variant *res);
extern void warning_mut(int cmd, variant *res);
extern void warts_mut(int cmd, variant *res);
extern void wasting_mut(int cmd, variant *res);
extern void weapon_skills_mut(int cmd, variant *res);
extern void weird_mind_mut(int cmd, variant *res);
extern void wings_mut(int cmd, variant *res);
extern void wraith_mut(int cmd, variant *res);

